In [1]:
from scipy import stats
import numpy as np

In [2]:
sample_A = np.array([78,84,92,88,75,80,85,90,87,79,78,84,92,88,75,80,85,90,87,79])
sample_B = np.array([82,88,75,90,78,85,88,77,92,80,82,88,75,90,78,85,88,77,92,80])

In [3]:
t_stats,p_val = stats.ttest_ind(sample_A,sample_B)

In [4]:
p_val

0.866845086698939

In [5]:
alpha=0.05

In [6]:
df=len(sample_A)+len(sample_B)-2

In [7]:
df

38

In [8]:
critical_t=stats.t.ppf(1-alpha/2,df)

In [9]:
critical_t

2.024394164575136

In [10]:
t_stats

0.16880536363378446

With T-value, of 0.168 is less than the critical value of 2.024. Therefore, significant difference is found between the exam scores of teaching method A and B.

With P-value, of 0.866 is greater than the significant level of 0.05. Therefore,there is no evidence to reject the null hypothesis. Therefore, there is no significant difference between the two teaching methods.

In [11]:
math1 = np.array([4,4,7,16,20,11,13,9,11,15])
math2=np.array([15,16,14,14,22,22,23,18,18,19])

In [12]:
t_stats,p_val = stats.ttest_rel(math1,math2)

In [13]:
t_stats

-4.953488372093023

In [14]:
p_val

0.0007875235561560145

In [15]:
alpha=0.05

In [16]:
df=len(math2)-1

In [17]:
critical_t=stats.t.ppf(1-alpha/2,df)

In [18]:
critical_t

2.2621571627409915

In [19]:
pop_mean=45

In [20]:
sample_mean=75
sample_std=25
sample_size=25

In [22]:
t_stat=(sample_mean -pop_mean )/(sample_std/np.sqrt(sample_size))

In [23]:
t_stat

6.0

In [24]:
df=sample_size-1

In [25]:
alpha=0.05

In [26]:
critical_t=stats.t.ppf(1-alpha,df)

In [27]:
critical_t

1.7108820799094275

In [28]:
p_val =1-stats.t.cdf(t_stat,df)

In [29]:
p_val

1.703654035845048e-06

Reject Ho, there is a significant diffeence between the weight before and after the camp.